cql3/expr/prepare_expr.cc

```diff
@@ -873,8 +874,26 @@ try_prepare_expression(const expression& expr, data_dictionary::database db, con
         [&] (const binary_operator& binop) -> std::optional<expression> {
             return prepare_binary_operator(binop, db, *schema_opt);
         },
-        [&] (const conjunction&) -> std::optional<expression> {
-            on_internal_error(expr_logger, "conjunctions are not yet reachable via prepare_expression()");
+        [&] (const conjunction& conj) -> std::optional<expression> {
+            std::vector<expression> prepared_children;
+            for (const expression& child : conj.children) {
+                std::optional<expression> prepared_child = try_prepare_expression(child, db, keyspace, schema_opt, receiver);
+                if (!prepared_child.has_value()) {
+                    return std::nullopt;
+                }
+                data_type prepared_child_type = type_of(*prepared_child);
+                if (prepared_child_type != boolean_type) {
+                    expression::printer child_printer {
+                        .expr_to_print = *prepared_child,
+                        .debug_mode = false,
+                    };
+                    throw exceptions::invalid_request_exception(format(
+                        "AND can only be applied to boolean conjunction values, but {} is of type {}",
+                        child_printer, prepared_child_type->name()));
+                }
+                prepared_children.push_back(std::move(*prepared_child));
+            }
+            return conjunction{std::move(prepared_children)};
```

Comment:
Since we do it elsewhere, check if all of the inputs are constants (and then you can compute it). Also you can drop any constant TRUE.

I'm not sure if it's okay for us the short-circuit if we have a NULL or FALSE. Probably it is okay.